lib/asset_sync/storage.rb

```diff
@@ -91,7 +91,13 @@ def get_local_files
       end
       log "Using: Directory Search of #{path}/#{self.config.assets_prefix}"
       Dir.chdir(path) do
-        to_load = self.config.assets_prefix.present? ? "#{self.config.assets_prefix}/**/**" : '**/**'
+        # So we may want to set the prefix something like 'prefix/'
+        # to prevent fuzzy folder matching in get_remote_files
+        # but that will cause issues with the folder expansion below in to_load
+        # So here we check if the prefix ends with a '/', and remove it as needed before
+        # looking for files.
+        path_prefix = self.config.assets_prefix =~ /\/$/ ? self.config.assets_prefix.sub(/\/$/, '') : self.config.assets_prefix
```

Comment:
You don't even have to do the check. If `.sub` can't find a traiing slash it just won't substitute anything. 
Might make the code easier to read.